plugins/languages/cmake-tools.nix

```diff
@@ -0,0 +1,391 @@
+                }; # a function that gets overseer.Task when it is created, before calling `task:start`
+              };
+
+              terminal = {
+                name = "Main Terminal";
+                prefix_name = "[CMakeTools]: "; # This must be included and must be unique, otherwise the terminals will not work. Do not use a simple spacebar " ", or any generic name
+                split_direction = "horizontal"; # "horizontal", "vertical"
+                split_size = 11;
+
+                # Window handling
+                single_terminal_per_instance = true; # Single viewport, multiple windows
+                single_terminal_per_tab = true; # Single viewport per tab
+                keep_terminal_static_location = true; # Static location of the viewport if available
+
+                # Running Tasks
+                start_insert = false; # If you want to enter terminal with :startinsert upon using :CMakeRun
+                focus = false; # Focus on terminal when cmake task is launched.
+              };
+            }
+          ''
+          ''
+            A list of default and possible values for runner.
+          '';
+      };
+
+    cmake_notifications =
+      helpers.mkCompositeOption ''
+        Notification settings.
+      ''
+      {
+        runner = helpers.mkCompositeOption "" {
+          enabled = helpers.defaultNullOpts.mkBool true "";
+        };
+
+        executor = helpers.mkCompositeOption "" {
+          enabled = helpers.defaultNullOpts.mkBool true "";
+        };
+
+        spinner = helpers.defaultNullOpts.mkListOf lib.types.str ''[ "⠋" "⠙" "⠹" "⠸" "⠼" "⠴" "⠦" "⠧" "⠇" "⠏" ]'' ''
+          Icons used for progress display.
+        '';
+
+        refresh_rate_ms = helpers.defaultNullOpts.mkPositiveInt 100 ''
+          How often to iterate icons.
+        '';
+      };
+  };
+
+  settingsExample = {
+    cmake_command = "cmake";
+    ctest_command = "ctest";
+    cmake_regenerate_on_save = true;
+    cmake_generate_options = {"-DCMAKE_EXPORT_COMPILE_COMMANDS" = 1;};
+    cmake_build_options = {};
+    cmake_build_directory = "out/\${variant:buildType}";
+    cmake_soft_link_compile_commands = true;
+    cmake_compile_commands_from_lsp = false;
+    cmake_kits_path = null;
+
+    cmake_variants_message = {
+      short = {show = true;};
+      long = {
+        show = true;
+        max_length = 40;
+      };
+    };
+
+    cmake_dap_configuration = {
+      name = "cpp";
+      type = "codelldb";
+      request = "launch";
+      stopOnEntry = false;
+      runInTerminal = true;
+      console = "integratedTerminal";
+    };
+
+    cmake_executor = {
+      name = "quickfix";
+      opts = {};
+      default_opts = {
+        quickfix = {
+          show = "always"; # "always", "only_on_error"
+          position = "belowright"; # "vertical", "horizontal", "leftabove", "aboveleft", "rightbelow", "belowright", "topleft", "botright", use `:h vertical` for example to see help on them
```

Comment:
comments in `settingsExample` won't show up in the docs. Most of these should probably be converted to enum types of moved to description text as appropriate, unless they're meant as internal comments for nixvim maintainers.

Reply:
Thanks for the review; these comments are copied from the default cmake-tools configuration upstream. I wasn´t aware that they wouldn't show up in the docs - then there is no point leaving them for nixvim maintainers. I'll just wait for further input to your other comments before proceeding, and accordingly remove them or move them to option descriptions.

Reply:
Personally, I think we should use enum options when upstream document a list of acceptable values.

Unless that list is likely to change frequently; that'd become a maintainance burden.

Reply:
> Personally, I think we should use enum options when upstream document a list of acceptable values.
> 
> Unless that list is likely to change frequently; that'd become a maintainance burden.

100% agree.